*-pc-windows-gnu target CI is broken

GitHub Actions for x86_64-pc-windows-gnu and i686-pc-windows-gnu are failing again. The following error message occurs during the `Set Up MinGW` step of the CI workflow:

```
Remove-Item: D:\a\_temp\9873c7c8-4f3e-4a8c-9439-180e50978186.ps1:140
Line |
 140 |          Remove-Item (Join-Path $mingw_lib 'libpthread.dll.a')
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path
     | 'C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw64\x86_64-w64-mingw32\lib\libpthread.dll.a' because it
     | does not exist.

Error: Process completed with exit code 1.
```

Examples of failed CI can be found in the #34 PR

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -53,18 +53,20 @@ jobs:
       # Install linux deps
       # - if: matrix.target.triple == 'x86_64-unknown-linux-musl'
       #   run: sudo apt-get install musl-tools
-            # Windows mingw 64bit
+      # Windows mingw 64bit
       - if: matrix.target.triple == 'x86_64-pc-windows-gnu'
         name: Set up MinGW
         uses: egor-tensin/setup-mingw@v2
         with:
           platform: x64
+          version: 12.2.0
       # Windows mingw 32bit
       - if: matrix.target.triple == 'i686-pc-windows-gnu'
         name: Set up MinGW
         uses: egor-tensin/setup-mingw@v2
         with:
           platform: x86
+          version: 12.2.0
       
       - name: Cargo Check - Default Features Only
         run: ${{ matrix.target.rustflags }} cargo check --target ${{ matrix.target.triple }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,9 +17,9 @@ jobs:
             triple: x86_64-pc-windows-msvc
           # Use because of broken GH Action
           # https://github.com/egor-tensin/setup-mingw/issues/6#issuecomment-1215265290
-          - os: windows-2019
+          - os: windows-latest
             triple: i686-pc-windows-gnu
-          - os: windows-2019
+          - os: windows-latest
             triple: x86_64-pc-windows-gnu
           
           - os: ubuntu-latest
@@ -59,15 +59,14 @@ jobs:
         uses: egor-tensin/setup-mingw@v2
         with:
           platform: x64
-          version: 12.2.0
+          version: 12.2.0 # https://github.com/egor-tensin/setup-mingw/issues/14
       # Windows mingw 32bit
       - if: matrix.target.triple == 'i686-pc-windows-gnu'
         name: Set up MinGW
         uses: egor-tensin/setup-mingw@v2
         with:
           platform: x86
-          version: 12.2.0
-      
+          version: 12.2.0 # https://github.com/egor-tensin/setup-mingw/issues/14      
       - name: Cargo Check - Default Features Only
         run: ${{ matrix.target.rustflags }} cargo check --target ${{ matrix.target.triple }}
       
@@ -79,3 +78,4 @@ jobs:
       
       - name: Cargo Tests - Nightly
         run: ${{ matrix.target.rustflags }} cargo +nightly test --target ${{ matrix.target.triple }} --all-features
+

